### PR TITLE
ODC-7639: Remove alerts list from dev console codebase

### PR DESCRIPTION
**Pull request by PeterYurkovich**
This PR is apart of a pair of PR's (along with [this](https://github.com/openshift/monitoring-plugin/pull/125) one in the monitoring-plugin) which look to move the alerting list page from the console codebase over to the monitorin-plugin. This PR needs to wait for the other PR in the monitoring-plugin codebase to merge before it can merge. So long as the other PR merges first, then in the case of a desync of releases there would be 2 alerting tabs which both link to the current developer perspective table.

This set of PR's change the UI of the developer perspective to be consistent with the admin perspective. The overall functionality stays the same, but there are some changes to the workflows required. For example to silence the alert you can't do so directly from the table anymore, instead you will click the ellipsis and then select the "Silence alert" option.

Screenshot 2024-06-27 at 1 36 14 PM

In addition, the tab control is currently done manually in the console codebase, with new additions being added at the end. A future PR will add a `console.action/group` as well as some logic to dictate the tab ordering inside the console codebase

### CODE REVIEW

**Comment by PeterYurkovich** [comment]
After the other PR is merged, I will un-hold this one

File: frontend/packages/dev-console/src/components/monitoring/__tests__/MonitoringPage.spec.tsx
Comment (jgbernalp) [inline frontend/packages/dev-console/src/components/monitoring/__tests__/MonitoringPage.spec.tsx:51]:
```suggestion
  it('should not render the Silences tab if user has no access to get prometheousRule resource', () => {
```

**Review by jgbernalp — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/monitoring/__tests__/MonitoringPage.spec.tsx
Comment (PeterYurkovich) [reply frontend/packages/dev-console/src/components/monitoring/__tests__/MonitoringPage.spec.tsx]: Fixed in latest commit

**Comment by juzhao** [comment]
https://github.com/openshift/monitoring-plugin/pull/125 is merged
/hold cancel

**Comment by PeterYurkovich** [comment]
Approval:
/assign @vikram-raj
Docs Approver: 
/assign @opayne1
PX Approver:
/assign @RickJWagner

**Review by vikram-raj — COMMENTED** [review]
Getting error `No Alert found` error when I try to see the Alert details page. 


https://github.com/openshift/console/assets/2561818/f076414d-9390-444f-80d8-d5c10e6b2437

**Comment by PeterYurkovich** [comment]
The monitoring-plugin was adding the rules labels onto the alerts labels, so when the labels got encoded into the url there were additional labels from the rules, causing the alerts detail page to fail to match.

I've prevented the rules labels from being added onto the alerts in the developer perspective in a separate for the monitoring-plugin [PR](https://github.com/openshift/monitoring-plugin/pull/132) and will update this one once that has merged

**Comment by PeterYurkovich** [comment]
The monitoring-plugin PR has been merged and the `No Alert found` issue has been fixed with it

**Review by vikram-raj — COMMENTED** [review]
/approve

**Comment by vikram-raj** [comment]
@PeterYurkovich you need to create and link CONSOLE or ODC jira ticket to get it merged.

**Comment by PeterYurkovich** [comment]
Based on the tide results I believe this PR is waiting on the qe-approved tag from @juzhao once they finish their testing. Is there another reason, such as tracking, that another ticket would be needed?

**Comment by jgbernalp** [comment]
@vikram-raj  it seems we already have the `jira/valid-reference`. Do we still need to create a jira ticket on those projects?

**Comment by vikram-raj** [comment]
> @vikram-raj it seems we already have the `jira/valid-reference`. Do we still need to create a Jira ticket on those projects?

Yes, CONSOLE / ODC ticket is required to get it in. As CONSOLE / ODC ticket is not attached to it so bot is looking for jira/valid-bug label. once you attach CONSOLE / ODC ticket then it will gone and the tide will be green as you have all the required labels.

**Comment by jgbernalp** [comment]
/retitle ODC-7639: Remove alerts list from dev console codebase

**Pull request merged**
